pyccel/ast/internals.py

```diff
@@ -292,8 +293,10 @@ def max_precision(objs : list, dtype = None, allow_native = True):
         return max(def_prec if o.precision == -1 \
                 else o.precision for o in objs if o.dtype is dtype)
     else:
-        return max(default_precision[str(o.dtype)] if o.precision == -1 \
-                else o.precision for o in objs)
+        if any(getattr(o, 'is_ndarray', False) for o in objs):
+            return get_final_precision(max(objs, key=attrgetter('precision')))
+        else:
+            return max(get_final_precision(o) for o in objs)
```

Comment:
I think these two statments are the same, so maybe you should just remove the if else?